numpy/_core/src/multiarray/iterators.c

```diff
@@ -881,13 +884,16 @@ iter_ass_subscript(PyArrayIterObject *self, PyObject *ind, PyObject *val)
     }
 
     if (index_type == HAS_BOOL) {
+        // This cannot be reached, because boolean indices temporarily raise an error.
+        // TODO: Enable valid boolean indices
```

Comment:
```suggestion
```
Also above, can be reached if an array is passed of course.  But can just apply that, and it might be th eonly thing.

Reply:
Whoops, you're right!